### PR TITLE
UX improvements, stacked support with arc land & cleanup after landing

### PR DESCRIPTION
**Pull request by gregmagolan**
Testing manually with phlq-test-1 repo

### CODE REVIEW

**Comment by gregmagolan** [comment]
Going to try to integration with ```protected function reconcileLocalState(
    $into_commit,
    ArcanistRepositoryLocalState $state) {
    $api = $this->getRepositoryAPI();
    $log = $this->getWorkflow()->getLogEngine();
    // Try to put the user into the best final state we can. This is very
    // complicated because users are incredibly creative and their local
    // branches may, for example, have the same names as branches in the
    // remote but no relationship to them.
    // First, we're going to try to update these local branches:
    //
    //   - the branch we started on originally; and
    //   - the local upstreams of the branch we started on originally; and
    //   - the local branch with the same name as the "into" ref; and
    //   - the local branch with the same name as the "onto" ref.
    //
    // These branches may not all exist and may not all be unique.
    //
    // To be updated, these branches must:
    //
    //   - exist;
    //   - have not been deleted; and
    //   - be connected to the remote we pushed into.```
to see if that will still do the right thing before going the custom route

**Comment by gregmagolan** [comment]
reconcileLocalState is too tied to the flow of arc doing landing locally. opting for a simplified PHLQ specific flow for cleanup after successful land and no-op on failed land.

**Comment by gregmagolan** [comment]
Also added support for stack diffs with arc land & fixed tailing of logs on retries

File: src/workflow/ArcanistLandWorkflow.php [hunk 1]
```diff
@@ -296,16 +296,23 @@ public function runWorkflow() {
...
     $repository_api = $working_copy->getRepositoryAPI();
 
+    $log = $this->getLogEngine();
+    $phlq_uri = $this->getPhlqUri();
+    $use_phlq = $this->getUsePhlq();
+
+    if ($use_phlq && empty($phlq_uri)) {
+      $log->writeWarning(
+        pht('PHLQ'),
+        pht("You have PHLQ enabled, but URI is missing. Attempting at landing by git push."));
```
Comment (alexeagle) [inline src/workflow/ArcanistLandWorkflow.php:306]: missing from where? maybe "this repository doesn't have phlq configured in .arcconfig"

**Review by alexeagle — APPROVED** [review]

File: src/config/arc/ArcanistArcConfigurationEngineExtension.php [hunk 2]
```diff
@@ -168,6 +168,16 @@ public function newConfigurationOptions() {
         ->setSummary(
           pht(
             'Error message when attempting to land a non-accepted revision.')),
+      id(new ArcanistStringConfigOption())
+        ->setKey('arc.land.buildfailures.message')
+        ->setDefaultValue(
+          pht(
+            'Rejected: You should not land revisions with failed or ongoing builds. '.
+            'If you know what you are doing and still want to land, add a '.
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply src/config/arc/ArcanistArcConfigurationEngineExtension.php:176]: That is just the default message, we'll update monorepo with a better one in .arcconfig 'arc.land.buildfailures.message' field

**Comment by gregmagolan** [comment]
Re-opening as source repo was deleted & recreated.

**Pull request closed without merging**